1-Draft/RFC00XX-Update-Error-View.md

```diff
@@ -0,0 +1,182 @@
+The addition of a “SimpleView” as default that contains only specific error
+information producing a single line on screen and in logs will improve customer success.
+
+Comprehensive error information is available to the customer through
+accessing the error object. For new and occasional customers,
+a cmdlet “Resolve-ErrorRecord” will simplify the process of getting detailed error information, improving customers success.
+
+## Specification
+
+The proposal is to add a new single-line default error message view called SimpleView
+and add a new cmdlet (Resolve-ErrorRecord) to provide detailed error information.
+
+__Key Design Considerations__
+
+1. To reduce confusion and improve debugging success,
+error messages, by default using WriteErrorLine, should produce a single line of text, including the word “ERROR:” to make consistent with Verbose and Warning messages. This is added to $errorView as view named "SimpleView". Conditionally, positional information will be added if from a script.
+
+- $ErrorView should contain these views
+
+    + NormalView
+    + CategoryView
+    + SimpleView
+
+- From Interactive: Error Message syntax for SimpleView
+
+```powershell
+ERROR: <Exception Message>
+ERROR: Cannot find path ‘C:\blah’ because it does not exist:
+```
+
+- From Script: Error Message syntax for SimpleView
+
+```powershell
+ERROR: <Exception Message>: <Positional Info>
+ERROR: Cannot find path ‘C:\blah’ because it does not exist: At line:1 char:1
+```
+
+2. A new cmdlet “Resolve-ErrorRecord” will produce detailed error information similar to Format-List * -Force.
+
+- Resolve-ErrorRecord will provide the following:
+
+    + Display Last Error ($Error[0]) – default behavior
+    + Accept Pipeline input – support $error[1] |
+    + Option for the Newest X errors in the session
+
+- Resolve-ErrorRecord syntax
+
+```powershell
+Resolve-ErrorRecord  [-InputObject <psobject>] [-Newest <Int32>] [-All] [<CommonParameters>]
+```
+
+First parameter set
+
+- Newest
+
+    + Datatype: int32
+    + specifies one or more of last errors to display
+    + Not required
+
+- All
+
+    + Datatype: Switch
+    + Specifies display all errors in session
+    + Not required
+
+Examples
+
+```powershell
+# Displays details of the last error displayed
+Resolve-ErrorRecord
+
+# Displays error details from pipeline error object
+$error[1] | Resolve-ErrorRecord
+
+# Display detailed error information for the last 5 errors
+Resolve-ErrorRecord -Last 5
+
+# Display detailed error information for the first 3 errors
+Resolve-ErrorRecord -First 3
+
+# Display detailed information for all errors that occured in the session
+Resolve-ErrorRecord -All
+```
+__Example 1__
+
+```powershell
+PS C:\test> Get-item c:\blah
+Get-item : Cannot find path 'C:\blah' because it does not exist. At line:1 char:1
+
+PS C:\test> Resolve-ErrorRecord
```

Comment:
Should have example where there are several nested exceptions.  This is common and manually digging into each one is unnecessary if we can automate it.